src/Datadog.Trace/Services/ThreadLocalNewRandomProvider.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+using System.Threading;
+using Datadog.Trace.Interfaces;
+
+namespace Datadog.Trace.Services
+{
+    internal class ThreadLocalNewRandomProvider : IRandomProvider
```

Comment:
Are we using this anywhere? Seems like even a static `Random` (shared process-wide across threads) would be preferable over this implementation. If it's not used, feel free to remove it.

Reply:
Yea, see below - I only left the extra impelmentations for now so you could run/test/view locally differences if you want, once we decide on the implementation we want to use, I'll get rid of the others